### PR TITLE
Remove openshift-sdn from the openshift binary

### PR DESCRIPTION
**Pull request by smarterclayton**
In preparation for dropping RPMs, openshift-sdn needs to be split out and a few dependencies to internal types snipped.  The new binary can then in the future be moved to another repo.  This removes one more piece from `openshift start`.  The new binary is `openshift-sdn` and it has the same behavior and flags as `openshift start network`

### CODE REVIEW

**Comment by danwinship** [comment]
This will presumably re-break dind but that should be an easy fix. poke @dcbw since he just fixed it and would know what needs to be done

File: images/node/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,17 @@
 #
-# This is an OpenShift node image with integrated OpenvSwitch SDN.
-#
-# This image expects to have a volume mounted at /etc/origin/node that contains
-# a KUBECONFIG file giving the node permission to talk to the master and a
-# node configuration file.
+# This is the OpenShift OpenvSwitch SDN image.
 #
 # The standard name for this image is openshift/origin-node
```
Comment (danwinship) [inline images/node/Dockerfile]: If this comment is still true it seems wrong, and probably both the image and this directory should be renamed?

Comment (smarterclayton) [reply images/node/Dockerfile]: Still debating how we're going to change this.  A new image is very expensive organizationally, so we're probably going to go stepwise here

**Comment by smarterclayton** [comment]
This is ready for review

File: cmd/openshift-sdn/openshift-sdn.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"math/rand"
+	"os"
+	"time"
+
+	"k8s.io/apiserver/pkg/util/logs"
+
+	"github.com/openshift/origin/pkg/cmd/flagtypes"
```
Comment (deads2k) [inline cmd/openshift-sdn/openshift-sdn.go]: @juanvallejo you're familiar with cobra bits. Can you see if this is a thing we need to move to library-go?  We need to see who is using it and what for.  We may just be able to kill all other consumers

File: pkg/cmd/server/start/network/start_network.go [hunk 3]
```diff
@@ -155,9 +161,9 @@ func (o NetworkOptions) RunNetwork() error {
 			nodeConfig.NodeName = o.NodeArgs.NodeName
 		}
 		nodeConfig.MasterKubeConfig = o.NodeArgs.KubeConnectionArgs.ClientConfigLoadingRules.ExplicitPath
-		validationResults = validation.ValidateInClusterNodeConfig(nodeConfig, nil)
+		//validationResults = validation.ValidateInClusterNodeConfig(nodeConfig, nil)
```
Comment (deads2k) [inline pkg/cmd/server/start/network/start_network.go]: This looks suspicious.  Why was it removed?

File: pkg/cmd/server/start/start.go [hunk 4]
```diff
@@ -27,10 +27,8 @@ func NewCommandStart(basename string, out, errout io.Writer, stopCh <-chan struc
 	}
 
 	startMaster, _ := NewCommandStartMaster(basename, out, errout)
-	startNodeNetwork, _ := NewCommandStartNetwork(basename, out, errout)
 	startEtcdServer, _ := openshift_etcd.NewCommandStartEtcdServer(openshift_etcd.RecommendedStartEtcdServerName, basename, out, errout)
 	cmds.AddCommand(startMaster)
-	cmds.AddCommand(startNodeNetwork)
```
Comment (deads2k) [inline pkg/cmd/server/start/start.go]: yay

**Comment by deads2k** [comment]
Just the question about removing validation.  I wouldn't lose the functionality like that for binary size. Seems like it would take care of itself as they switch to their own config.

**Comment by deads2k** [comment]
/lgtm
/hold

holding pending the validation question.

File: cmd/openshift-sdn/openshift-sdn.go (hunk 2 above)
Comment (juanvallejo) [reply cmd/openshift-sdn/openshift-sdn.go]: That package is used by the following files in Origin:

```
pkg/oc/cli/login/login.go
pkg/oc/cli/cli.go
pkg/cmd/server/start/start_controllers.go
pkg/cmd/server/start/master_args.go
pkg/cmd/server/start/start_api.go
pkg/cmd/server/start/kube_connection_args.go
pkg/cmd/server/start/listen_arg.go
pkg/cmd/server/admin/create_nodeconfig.go
pkg/cmd/server/kubernetes/master/master_config.go
pkg/cmd/openshift/openshift.go
```

The package allows us to bind entire structs to cobra flags, and handle how flag values are set into fields in that struct. Not sure that cobra has a built-in solution for this, other than providing the `Value` interface.
We also use it in `openshift.go` and `cli.go` to bind `glog` flags to a given flagset.

**Comment by smarterclayton** [comment]
Forked the node-config validation into a subdir and remove anything that wasn't related to networking from that fork.

File: cmd/openshift-sdn/openshift-sdn.go (hunk 2 above)
Comment (deads2k) [reply cmd/openshift-sdn/openshift-sdn.go]: @juanvallejo remember to come back to it once we kill openshift start

**Comment by smarterclayton** [comment]
Rebased and green, reapplying label

**Pull request merged**